test/runnable/test_cdstrpar.d

```diff
@@ -0,0 +1,266 @@
+        /* pop    rbp                     */ 0x5d,
+        /* ret                            */ 0xc3,
+        /* add    BYTE PTR [rax],al       */ 0x00, 0x00,
+    ]),
+    Code!(ubyte, 64)([
+        /* push   rbp                     */ 0x55,
+        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
+        /* sub    rsp,0x40                */ 0x48, 0x83, 0xec, 0x40,
+        /* lea    rdi,[rbp-0x40]          */ 0x48, 0x8d, 0x7d, 0xc0,
+        /* call   11 <testee_ubyte_64+0x11> */ 0xe8, 0x00, 0x00, 0x00, 0x00,
+        /* mov    rsi,rax                 */ 0x48, 0x89, 0xc6,
+        /* mov    ecx,0x8                 */ 0xb9, 0x08, 0x00, 0x00, 0x00,
+        /* add    rsi,0x38                */ 0x48, 0x83, 0xc6, 0x38,
+        /* push   QWORD PTR [rsi]         */ 0xff, 0x36,
+        /* sub    rsi,0x8                 */ 0x48, 0x83, 0xee, 0x08,
+        /* dec    ecx                     */ 0xff, 0xc9,
+        /* jne    1d <testee_ubyte_64+0x1d> */ 0x75, 0xf6,
+        /* call   2c <testee_ubyte_64+0x2c> */ 0xe8, 0x00, 0x00, 0x00, 0x00,
+        /* add    rsp,0x40                */ 0x48, 0x83, 0xc4, 0x40,
+        /* mov    rsp,rbp                 */ 0x48, 0x8b, 0xe5,
+        /* pop    rbp                     */ 0x5d,
+        /* ret                            */ 0xc3,
+        /* add    BYTE PTR [rax],al       */ 0x00, 0x00,
+    ]),
+);
+
+// end
+// dfmt on
+
+bool matches(const(ubyte)[] code, const(ubyte)[] exp)
+{
+    assert(code.length == exp.length);
+    foreach (ref i; 0 .. code.length)
+    {
+        if (code[i] == exp[i])
+            continue;
+        // wildcard match for relative call displacement
+        if (i && exp.length - (i - 1) >= 5 && exp[i - 1 .. i + 4] == [0xe8, 0x00, 0x00, 0x00, 0x00])
+        {
+            i += 3;
+            continue;
+        }
+        return false;
+    }
+    return true;
+}
+
+alias testCases = AliasSeq!(baselineCases);
+
+void main()
+{
+    foreach (tc; testCases)
+    {
+        auto code = (cast(ubyte*)&testee!(tc.T, tc.N))[0 .. tc.code.length];
+        bool failure;
+        if (!code.matches(tc.code))
+        {
+            fprintf(stderr, "Expected code sequence for testee!(%s, %u) not found.",
+                    tc.T.stringof.ptr, tc.N);
+            fprintf(stderr, "\n  Expected:");
+            foreach (i, d; tc.code)
+            {
+                if (tc.code[i] != code[i])
+                    fprintf(stderr, " \033[32m0x%02x\033[0m", d);
+                else
+                    fprintf(stderr, " 0x%02x", d);
+            }
+            fprintf(stderr, "\n    Actual:");
+            foreach (i, d; code)
+            {
+                if (tc.code[i] != code[i])
+                    fprintf(stderr, " \033[31m0x%02x\033[0m", d);
+                else
+                    fprintf(stderr, " 0x%02x", d);
+            }
+            fprintf(stderr, "\n");
+            failure = true;
+        }
+        assert(!failure);
+    }
```

Comment:
There's a lot of code duplication from `test_cdvecfill.d`. Any reason why this can't be shared?

Reply:
It's yet unclear how this could be made into a general framework to test codegen, or whether better tools already exist for that.
I've tried to keep them very similar to facilitate future generalization, but for the time being those are ad-hoc solutions and I wouldn't want to waste more time at generalizing sth. that's only used twice.

Reply:
Perhaps look into how LLVM/GCC are testing this.